content/news/this-week-in-fluvio-0051.md

```diff
@@ -0,0 +1,95 @@
+---
+title: "This Week in Fluvio #51"
+date: 2023-04-11
+weight: 20
+---
+Fluvio is a distributed, programmable streaming platform written in Rust.
+
+---
+Welcome to the 51st edition of this week in Fluvio.
+
+For this edition we have more details on **Community Contribution** 
+
+## Community update
+[Carson Rajcan](https://github.com/crajcan) presented to our team a really cool contribution to the Fluvio OSS.
+
+In the project Carson developed the functionality to apply Smart Module transformation for Producer - [PR #3014](crecdfhttps://github.com/infinyon/fluvio/pull/3014)
+
+Here is Carson's experience...
+
+## The Problem
+
+Get the Stream Processing Unit to apply Smart Module transformations for the Producer requests coming from the CLI (before commit). At present, Fluvio Smart Modules are only applied for Consumer requests after being read from disk.
+
+## Why This Problem?
+
+While investigating another issue, I had recently gained some insight into how the Stream Processing Unit handles Producer requests, as well as how it uses the Fluvio Smart Engine to transform records for the Consumer. Critically, I noticed that the Smart Engine was well encapsulated -- it was going to be easy to repurpose. Also that the Producer request handler was well organized, it wasn't going to be a nightmare to plug in some additional functionality.
+
+## The Workflow
+
+Where was I going to start? Well, TDD is my friend. There were plenty of test cases for Consumer Smart Module transformations that used the CLI. All I had to do was move the Smart Module options from the Consumer commands over to the Producer commands:
+
+```
+echo "foobar@test.com" | fluvio produce emails
```

Comment:
foobar@test.com -> FooBar@test.com

Then add a result to show
foobar@test.com

Thant way it's obvious what the SmartModule purpose is.